### PR TITLE
Added create job form and implemented s3 image upload functionality

### LINKED ISSUE #307: ✨ Improvement: add company image upload via s3
### 📜 Description

image

### 👟 Reproduction steps

_No response_

### 👍 Expected behavior

get the image uploaded to s3 and then save the link in db 

### 📃 Provide any additional context for the Bug.

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/code100x/job-board/blob/main/CONTRIBUTING.md)

### LINKED ISSUE #316: 🚀 Feature: Post a job form
### 🔖 Feature description

take inspiration from remotejob.com 

### 🎤 Why is this feature needed ?

to allow peoples to come and post jobs and gigs

### ✌️ How do you aim to achieve this?

by adding the form 

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/code100x/job-board/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by developingright**
fix #316 fix #307 

https://github.com/user-attachments/assets/1c0b8acc-46b6-49a8-8991-889fdf9b9973

- Changed Schema ( Added New Job fields )
- Added Zod Validation for new form fields
- Added fully functional create job form
- implemented s3 upload and image preview in job form
- Fixed job actions 
- Changed Seed file

### CODE REVIEW

**Comment by anuragnegi000** [comment]
@VineeTagarwaL-code please consider #316 PR in this

File: src/app/api/s3-upload/route.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { NextResponse } from 'next/server';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3';
+
+const s3Client = new S3Client({
```
Comment (VineeTagarwaL-code) [inline src/app/api/s3-upload/route.ts:5]: this is server component, why use NEXT_PUBLIC then  ?

**Review by VineeTagarwaL-code — COMMENTED** [review]
everything feels fine, please do the asked change then we will be good to go

**Comment by developingright** [comment]
@VineeTagarwaL-code I have made the required changes, please check.

**Review by VineeTagarwaL-code — APPROVED** [review]

**Pull request merged**